docs-core/src/main/java/com/sismics/docs/core/dao/UserDao.java

```diff
@@ -278,7 +279,22 @@ public void delete(String username, String userId) {
      * @return Hashed password
      */
     private String hashPassword(String password) {
-        return BCrypt.withDefaults().hashToString(10, password.toCharArray());
+        int bcryptWork = Constants.DEFAULT_BCRYPT_WORK;
+        String envBcryptWork = System.getenv(Constants.BCRYPT_WORK_ENV);
+        if (envBcryptWork != null) {
+            try {
+                int envBcryptWorkInt = Integer.parseInt(envBcryptWork);
+                if (envBcryptWorkInt >= 4 && envBcryptWorkInt <= 31) {
+                   bcryptWork = envBcryptWorkInt;
+                } else {
+                    //ToDo: Log warning? (out of scope)
```

Comment:
Log using slf4j logger (please see FileService.java for example) as "warning" level and fallback to default bcrypt work.